app/code/Magento/ImportExport/Model/Import/SampleFileProvider.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\ImportExport\Model\Import;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Component\ComponentRegistrar;
+use Magento\Framework\Filesystem\Directory\ReadInterface;
+
+/**
+ * Import Sample File Provider model.
+ * This class support only *.csv.
+ */
+class SampleFileProvider
+{
+    /**
+     * Associate an import entity to its module, e.g ['entity_name' => 'module_name']
+     * @var array
+     */
+    private $samples;
+
+    /**
+     * @var \Magento\Framework\Component\ComponentRegistrar
+     */
+    private $componentRegistrar;
+
+    /**
+     * @var \Magento\Framework\Filesystem\Directory\ReadFactory
+     */
+    private $readFactory;
+
+    /**
+     * @param \Magento\Framework\Filesystem\Directory\ReadFactory $readFactory
+     * @param ComponentRegistrar $componentRegistrar
+     * @param array $samples
+     */
+    public function __construct(
+        \Magento\Framework\Filesystem\Directory\ReadFactory $readFactory,
+        \Magento\Framework\Component\ComponentRegistrar $componentRegistrar,
+        array $samples = []
+    ) {
+        $this->readFactory = $readFactory;
+        $this->componentRegistrar = $componentRegistrar;
+        $this->samples = $samples;
+    }
+
+    /**
+     * Returns the Size for the given file associated to an Import entity
+     *
+     * @param string $entityName
+     * @throws NoSuchEntityException
+     * @return int
+     */
+    public function getSize(string $entityName): int
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
+        $filePath = $this->getPath($entityName);
+        $fileSize = isset($directoryRead->stat($filePath)['size'])
+            ? $directoryRead->stat($filePath)['size'] : null;
+
+        return $fileSize;
+    }
+
+    /**
+     * Returns Content for the given file associated to an Import entity
+     *
+     * @param string $entityName
+     * @throws NoSuchEntityException
+     * @return string
+     */
+    public function getFileContents(string $entityName): string
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
+        $filePath = $this->getPath($entityName);
+
+        return $directoryRead->readFile($filePath);
+    }
+
+    /**
+     * @param string $entityName
+     * @return string $entityName
+     * @throws NoSuchEntityException
+     */
+    private function getPath(string $entityName): string
+    {
+        $directoryRead = $this->getDirectoryRead($entityName);
```

Comment:
Underhood of this method, we create the directory, but in `getModuleName` we can throw exception
So maybe do we need to finish all of the checks before performing some actions?

Reply:
Done.